src/app/home-page/state/recent-events.service.ts

```diff
@@ -1,24 +1,36 @@
 import { Injectable } from '@angular/core';
 import { ID } from '@datorama/akita';
-import { Event, EventsService } from 'app/shared/openapi';
+import { Event, EventsService, User } from 'app/shared/openapi';
 import { finalize, tap } from 'rxjs/operators';
 import { RecentEventsStore } from './recent-events.store';
 
 @Injectable({ providedIn: 'root' })
 export class RecentEventsService {
   constructor(private recentEventsStore: RecentEventsStore, private eventsService: EventsService) {}
 
-  get() {
+  get(user?: User) {
     this.recentEventsStore.setLoading(true);
-    this.eventsService
-      .getEvents('ALL_STARRED')
-      .pipe(
-        finalize(() => this.recentEventsStore.setLoading(false)),
-        tap((allStarredEvents) => {
-          this.recentEventsStore.set(allStarredEvents);
-        })
-      )
-      .subscribe();
+    if (user) {
+      this.eventsService
+        .getUserEvents(user.id, 'PROFILE')
+        .pipe(
+          finalize(() => this.recentEventsStore.setLoading(false)),
+          tap((allStarredEvents) => {
+            this.recentEventsStore.set(allStarredEvents);
+          })
+        )
+        .subscribe();
+    } else {
+      this.eventsService
+        .getEvents('PROFILE')
+        .pipe(
```

Comment:
Nit-pick, this code block duplicates lines 16-22.

Reply:
This uses 'getEvents' rather than 'getUserEvents', the difference is getEvents gets it for the user that is logged in, so I don't have to pass on a user which was taken from the activatedRoute param

Reply:
However the 403 errors are concerning, I will run on my local and try to see why that's happening.

Reply:
For the duplication, I was referring to lines 16-22, and lines 26-32 -- they're identical, I think.

Even though you're calling different endpoints, they both return the same response type, so you don't need to repeat that block.

Something like, untested:

```
const events = user ? this.eventsService.getUser(user.id, 'PROFILE') : this.eventsService.getEvents('PROFILE');
events.pipe(
   finalize(...
...
```